File: cosmo/src/test/functional/protocol/test_cmp.py
#   Copyright (c) 2006-2007 Open Source Applications Foundation
#
#   Licensed under the Apache License, Version 2.0 (the "License");
#   you may not use this file except in compliance with the License.
#   You may obtain a copy of the License at
#
#       http://www.apache.org/licenses/LICENSE-2.0
#
#   Unless required by applicable law or agreed to in writing, software
#   distributed under the License is distributed on an "AS IS" BASIS,
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#   See the License for the specific language governing permissions and
#   limitations under the License.

import cosmoclient
import random

import cosmo_test_lib

def setup_module(module):
    cosmo_test_lib.setup_module(module)
    client = cosmoclient.CosmoClient(module.SERVER_URL)
    client.set_basic_auth(module.ADMIN_USER, module.ADMIN_PASS)
    module.client = client

def test_list_users():
    client.get(client._cmp_path+'/users')
    assert client.response.status == 200
    assert client.response.body.find('root') is not -1
    
def test_view_user():
    client.get(client._cmp_path+'/user/root')
    assert client.response.status == 200
    assert client.response.body.find('root') is not -1

# def test_create_user():
#     client.add_user(TEST_USER, TEST_PASS, TEST_FIRST_NAME, TEST_LAST_NAME, TEST_EMAIL)
#     assert client.response.status == 201
#     client.add_user(TEST_USER+"-2", TEST_PASS, TEST_FIRST_NAME, TEST_LAST_NAME, TEST_USER+"-2@osafoundation.org", request_method=client.post)
#     assert client.response.status == 201
#     
# def test_modify_user():
#     client.add_user('test_modify_user', 'test_pass', 'Test', 'Name', 'me@fake.com')
#     assert client.response.status == 201
#     client.modify_user({'username':'test_modify_user', 'email':'new_email@fake.com'})
#     assert client.response.status == 204
#     client.get(client._cmp_path+'/user/test_modify_user')
#     assert client.response.body.find('new_email@fake.com') is not -1
#     client.modify_user({'username':'test_modify_user', 'email':'new_new_email@fake.com'}, request_method=client.post)
#     assert client.response.status == 204
#     client.get(client._cmp_path+'/user/test_modify_user')
#     assert client.response.body.find('new_new_email@fake.com') is not -1
    
def test_delete_user():
    import uuid
    name = str(uuid.uuid1()).replace('-', '')
    client.add_user(name, 'test_pass', 'Test', 'Name', name+'@blah.com')
    assert client.response.status == 201
    client.remove_user(name)
    assert client.response.status == 204

def test_administrator_tag():
    from xml.etree import ElementTree
    
    client.get(client._cmp_path+'/users')
    assert client.response.status == 200
    assert client.response.body.find('root') is not -1
    root_user_element = [x for x in client.response.tree.findall('{http://osafoundation.org/cosmo/CMP}user') if x.find('{http://osafoundation.org/cosmo/CMP}username').text == 'root'][0]
    assert root_user_element.find('{http://osafoundation.org/cosmo/CMP}administrator').text == 'true'
    test_user_element = [x for x in client.response.tree.findall('{http://osafoundation.org/cosmo/CMP}user') if x.find('{http://osafoundation.org/cosmo/CMP}username').text == TEST_USER][0]
    assert test_user_element.find('{http://osafoundation.org/cosmo/CMP}administrator').text == 'false'
    client.modify_user({'username':TEST_USER, 'administrator':'true'})
    assert client.response.status == 204
    client.get(client._cmp_path+'/users')
    assert client.response.status == 200
    test_user_element = [x for x in client.response.tree.findall('{http://osafoundation.org/cosmo/CMP}user') if x.find('{http://osafoundation.org/cosmo/CMP}username').text == TEST_USER][0]
    assert test_user_element.find('{http://osafoundation.org/cosmo/CMP}administrator').text == 'true'
    client.modify_user({'username':TEST_USER, 'administrator':'false'})
    assert client.response.status  == 204
    
    
def teardown_module(module):
    client.remove_user('test_modify_user')
    assert client.response.status == 204
    client.remove_user(TEST_USER)
    assert client.response.status == 204  
    client.remove_user(TEST_USER+'-2')
    assert client.response.status == 204  

        
    

File: cosmo/src/test/unit/js/cosmo/cmp/test_cmp.js
/*
 * Copyright 2007 Open Source Applications Foundation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

dojo.provide("cosmotest.cmp.test_cmp")
dojo.require("cosmo.cmp");

cosmotest.cmp.test_cmp = {
    test_userXML: function(){
        var user1 = {
            username: "test1",
            password: "password1",
            firstName: "test1",
            lastName: "mctest1",
            email: "test1@example.com"
        };

        var user1XML = '<?xml version="1.0" encoding="utf-8" ?>\r\n' +
            '<user xmlns="http://osafoundation.org/cosmo/CMP">' +
            '<username>test1</username>' +
            '<password>password1</password>' +
            '<firstName>test1</firstName>' +
            '<lastName>mctest1</lastName>' +
            '<email>test1@example.com</email>' +
            '</user>';

        jum.assertEquals("user 1 wrong", user1XML, cosmo.cmp.userHashToXML(user1));

        var user2 = {
            username: "test2",
            password: "password2",
            firstName: "test2",
            lastName: "mctest2",
            email: "test2@example.com",
            subscription: {
                name: "test subscription",
                ticket: "09876",
                uuid: "0123456789"
            }
        };

        var user2XML = '<?xml version="1.0" encoding="utf-8" ?>\r\n' +
            '<user xmlns="http://osafoundation.org/cosmo/CMP">' +
            '<username>test2</username>' +
            '<password>password2</password>' +
            '<firstName>test2</firstName>' +
            '<lastName>mctest2</lastName>' +
            '<email>test2@example.com</email>' +
            '<subscription name="test subscription" ticket="09876">0123456789</subscription>' + 
            '</user>';

        jum.assertEquals("user 2 wrong", user2XML, cosmo.cmp.userHashToXML(user2));

        user2.subscription.name = null;

        try {
            cosmo.cmp.userHashToXML(user2);
            jum.assertTrue("null name didn't throw exception", false);
        } catch (e1){
            jum.assertTrue("e1 name not null", e1.name == null);
            jum.assertEquals("e1 ticket wrong", e1.ticket, "09876");
            jum.assertEquals("e1 uuid wrong", e1.uuid, "0123456789");
        }

        user2.subscription.name = "test subscription";
        user2.subscription.ticket = null;

        try {
            cosmo.cmp.userHashToXML(user2);
            jum.assertTrue("null ticket didn't throw exception", false);
        } catch (e2){
            jum.assertTrue("e2 ticket not null", e2.ticket == null);
            jum.assertEquals("e2 name wrong", e2.name, "test subscription");
            jum.assertEquals("e2 uuid wrong", e2.uuid, "0123456789");
        }

        user2.subscription.ticket = "09876";
        user2.subscription.uuid = null;

        try {
            cosmo.cmp.userHashToXML(user2);
            jum.assertTrue("null uuid didn't throw exception", false);
        } catch (e3){
            jum.assertTrue("e3 uuid not null", e3.uuid == null);
            jum.assertEquals("e3 name wrong", e3.name, "test subscription");
            jum.assertEquals("e3 ticket wrong", e3.ticket, "09876");
        }
    }
}


